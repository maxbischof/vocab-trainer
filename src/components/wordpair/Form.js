import React, { createRef, useState } from 'react'
import styled from 'styled-components'
import Input from '../ui/Input'
import Button from '../ui/Button'
import GrammarMenu from './GrammarMenu'

export default function Form({ wordPairs, setWordPairs }) {
  const [formInput, setFormInput] = useState({ foreign: '', native: '' })
  const [wordClass, setWordClass] = useState('N')
  const [gender, setGender] = useState()
  const [person, setPerson] = useState()
  const [number, setNumber] = useState()

  const foreignInput = createRef()

  function onSubmit(event) {
    event.preventDefault()
    const newWords = [...wordPairs]
    newWords.push(formInput)
    setWordPairs(newWords)
    setFormInput({ foreign: '', native: '' })
    foreignInput.current.focus()
  }

  function onChange(event) {
    const language = event.target.name
    const newFormInput = { ...formInput }
    newFormInput[language] = event.target.value
    setFormInput(newFormInput)
  }

  return (
    <StyledForm onSubmit={onSubmit}>
      <Input
        name="foreign"
        type="text"
        placeholder="Foreign"
        onChange={onChange}
        value={formInput.foreign}
        ref={foreignInput}
      />
      <Input
        name="native"
        type="text"
        placeholder="Native"
        onChange={onChange}
        value={formInput.native}
      />
      <GrammarMenu
        wordClass={wordClass}
        setWordClass={setWordClass}
        gender={gender}
        setGender={setGender}
        person={person}
        setPerson={setPerson}
        number={number}
        setNumber={setNumber}
      />
      <Button type="submit">Add word</Button>
    </StyledForm>
  )
}

const StyledForm = styled.form`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 10px;
`
