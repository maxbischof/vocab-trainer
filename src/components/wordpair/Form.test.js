import React from 'react'
import Form from './Form'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(<Form />).toJSON()
  expect(tree).toMatchSnapshot()
})
