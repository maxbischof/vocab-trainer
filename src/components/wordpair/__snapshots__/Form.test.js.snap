// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="sc-jrAGrp kfPisi"
  onSubmit={[Function]}
>
  <svg
    className="sc-kEjbxe dHxCwD"
    onClick={[Function]}
  >
    wordClass.svg
  </svg>
  <input
    className="sc-bdfBwQ hRYrgT"
    name="foreign"
    onChange={[Function]}
    placeholder="Foreign"
    type="text"
    value=""
  />
  <input
    className="sc-bdfBwQ hRYrgT"
    name="native"
    onChange={[Function]}
    placeholder="Native"
    type="text"
    value=""
  />
  <button
    className="sc-gsTCUz hwVgqy"
    type="submit"
  >
    Add word
  </button>
</form>
`;
